### PR TITLE
Don't override a user's cmd.prompt setting

### PR DESCRIPTION
**Pull request by jduck**
When using the debugger I'd like to set the prompt to something nice. However, I'm not a fan of "dbg.status"... Instead, stop setting the prompt if it's already set...

### CODE REVIEW

File: libr/core/io.c [hunk 1]
```diff
@@ -39,10 +41,15 @@ R_API int r_core_setup_debugger (RCore *r, const char *debugbackend, bool attach
...
 	}
 	r_core_cmd (r, "sr PC", 0);
-	if (r_config_get_i (r->config, "dbg.status"))
-		r_config_set (r->config, "cmd.prompt", ".dr*;drd;sr PC;pi 1;s-");
-	else
-		r_config_set (r->config, "cmd.prompt", ".dr*");
+
+	/* set the prompt if it's not been set already by the callbacks */
+	prompt = r_config_get (r->config, "cmd.prompt");
+	if (prompt && !strcmp (prompt, "")) {
```
Comment (radare) [inline libr/core/io.c]: && !*prompt

File: libr/core/io.c [hunk 2]
```diff
@@ -39,10 +41,15 @@ R_API int r_core_setup_debugger (RCore *r, const char *debugbackend, bool attach
...
+		if (r_config_get_i (r->config, "dbg.status"))
+			r_config_set (r->config, "cmd.prompt", ".dr*;drd;sr PC;pi 1;s-");
+		else
+			r_config_set (r->config, "cmd.prompt", ".dr*");
+	}
```
Comment (radare) [inline libr/core/io.c]: better refactor to something like this:

```c
prompt = r_config_get_i (r->config, "dbg.status") ? ".dr*;drd;sr PC;pi 1;s-": ".dr*";
r_config_set (r->config, "cmd.prompt", prompt);
```

Comment (jduck) [inline libr/core/io.c]: I'm really against ternary operator use, especially one this complicated. Further, this is totally irrelevant to the changeset.

File: libr/core/io.c (hunk 1 above)
Comment (jduck) [inline libr/core/io.c]: I initially thought the same but the existing code already did the strcmp with "" so I went with what I thought would be the path of least resistance. Imagine my surprise =) Happy to fix this one before PR though.

**Comment by radare** [comment]
thanks

**Pull request merged**